### PR TITLE
Update docs with branching strategy and compatibility matrix

### PR DESCRIPTION
**Pull request by a-blender**
Issue: https://github.com/rancher/terraform-provider-rancher2/issues/1052

### CODE REVIEW

**Review by HarrisonWAffel — APPROVED** [review]

**Review by snasovich — CHANGES_REQUESTED** [review]
I will have some suggestions for this later, so just marking with "Request changes" for now to block merging.

File: docs/compatability-matrix.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Rancher2 Provider
```
Comment (snasovich) [inline docs/compatability-matrix.md]: There is a typo in fule name

File: docs/compatability-matrix.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+### Version compatibility matrix
+
+#### Rancher 2.6
+
+| Terraform Version | 2.0.0         | 2.x.x  | 2.x.x         |
```
Comment (snasovich) [inline docs/compatability-matrix.md]: I think we want to transpose this table and list TF release versions as rows and list min/max Rancher versions as columns. With the current approach, we're adding a new column for each TF provider release which eventually will make a very wide table.

**Review by snasovich — CHANGES_REQUESTED** [review]
Couple initial comments to address, will need to review again later.

File: docs/compatability-matrix.md (hunk 2 above)
Comment (snasovich) [reply docs/compatability-matrix.md]: Discussed with @annablender offline and we've agreed on a more flexible approach where we list "reference" Racher<->TF provider pairs and make disclaimers of using combinations where TF provider version is higher than reference Rancher version and vice-versa.

File: README.md [hunk 3]
```diff
@@ -84,14 +84,29 @@ Due to [network limitation](https://docs.docker.com/docker-for-mac/networking/#k
 $ EXPOSE_HOST_PORTS=true make docker-testacc
 ```
 
+Branching the Provider
+---------------------------
+
+The provider is branched into two release lines that have major version alignment with Rancher 2.6 and 2.7. The `release/v2` branch with 2.0.0+ is aligned with Rancher 2.6 and `master` with 3.0.0+ is aligned with Rancher 2.7. Terraform fixes and new features will be available on `master` but only bug fixes will be backported to `release/v2.`
```
Comment (snasovich) [inline README.md]: Don't want to necessarily commit to backporting all fixes, so adding a bit of a disclaimer. Also minor formatting fix.

```suggestion
The provider is branched into two release lines that have major version alignment with Rancher 2.6 and 2.7. The `release/v2` branch with 2.0.0+ is aligned with Rancher 2.6 and `master` with 3.0.0+ is aligned with Rancher 2.7. Terraform provider fixes and new features will be available on `master` but only bug fixes will be backported to `release/v2` as necessary.
```

File: README.md [hunk 4]
```diff
@@ -84,14 +84,29 @@ Due to [network limitation](https://docs.docker.com/docker-for-mac/networking/#k
...
+
+Aligning major provider releases with minor Rancher releases means,
+
+* We can follow semver 
+* We can cut patch/minor versions on an as-needed basis to fix bugs or add new resources 
+* We have 'out of band' flexibility and are only tied to releasing a new version of the provider when we get a new 2.x Rancher minor version.
+
+See the [compatibility matrix](https://github.com/rancher/terraform-provider-rancher2/tree/master/docs/compatibility-matrix.md) for details.
```
Comment (snasovich) [inline README.md]: Suggesting to use relative link (https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes#relative-links-and-image-paths-in-readme-files) here as it's more resilient to changes to URLs and will work if we backport this docs change to another branch without needing to change it (unless we don't want to port actual `compatibility-matrix.md` file and always point to it in the master 🤷 ).

File: docs/compatibility-matrix.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+#### Rancher 2.7
+
+| Terraform provider version | Rancher |
+|----------------------------------------|:-------:|
+| 3.0.0                                  | 2.7.2   |
+| 3.0.1 (3.1.0 for features)             | 2.7.x   |
+
+**Can I use an earlier Terraform version?** Yes, but Terraform may not support all features and fields supported in your Rancher version so provisioning may be limited.
+
+**Can I use a later Terraform version?** Yes, but you must NOT use any new features and fields that your Rancher version does not support.
```
Comment (snasovich) [inline docs/compatibility-matrix.md:23]: Nice, I like this wording!

Maybe add a section like "FAQ" above it so these remarks are technically in their own section and not with "Rancher 2.7"?

File: docs/compatibility-matrix.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+### Version compatibility matrix
+
+The version matrix specifies the Terraform provider version _recommended_ to use with the associated minor Rancher version that it was released for. When updating the version matrix, add a row for each Terraform version released with a minor Rancher release.
+
+#### Rancher 2.6
+
+| Terraform provider version | Rancher |
+|----------------------------------------|:-------:|
+| 2.0.0                                  | 2.6.11  |
+| 2.0.1 (2.1.0 for features)             | 2.6.x   |
```
Comment (snasovich) [inline docs/compatibility-matrix.md]: Recommending to not list not-yet-released versions, only the ones that we already released. I understand it's an example, but don't want it to be misleading. The same applies to the 2.7 table below.

**Review by snasovich — CHANGES_REQUESTED** [review]
Looks good, some minor comments to address and it should be good to go from my point of view.

File: docs/compatibility-matrix.md (hunk 6 above)
Comment (a-blender) [reply docs/compatibility-matrix.md]: Do you mean to remove the entire second row here or just modify to list

`2.0.x (2.x.0 for features)`

Comment (snasovich) [reply docs/compatibility-matrix.md]: Just remove, we explain the approach to versioning in Readme already. This matrix is for released(to be released) versions.

**Review by snasovich — APPROVED** [review]

**Pull request merged**